lib/ipam/ipam.go

```diff
@@ -1383,6 +1344,62 @@ func (c ipamClient) convertBackendToIPAMConfig(cfg *model.IPAMConfig) *IPAMConfi
 	}
 }
 
+// maybeReleaseAffinity retrieves the pool and node for the given block and determines
+// if the pool still selects node. If it no longer matches, it will release the block
+// affinity for that node.
+// Returns a bool indicating if the block was released.
+func (c ipamClient) maybeReleaseAffinity(ctx context.Context, b *model.AllocationBlock) (bool, error) {
```

Comment:
I know I suggested this name, but I think a more accurate name would be `ensureConsistentAffinity` or something like that. This function is meant to release affinity for a specific scenario, so I'd like to make that clear in the name.